### ISSUE TITLE
tarp fails to parse binary when compiled with RUSTFLAGS='-C link-arg=-fuse-ld=lld'

### ISSUE THREAD

**Issue by yaahc** [xd009642/tarpaulin#286]
I'm trying to use tarpaulin to get coverage but its failing to parse the binary. I thought it might be something about my project so I made a new lib with nothing other than the default boilerplate and its still giving me the same error. Running with the -v flag doesn't give me more info.

```
tmp with odzt60a
❯ cargo new --lib covtest
     Created library `covtest` package

tmp with odzt60a
❯ cd covtest

tmp/covtest is 📦 v0.1.0 via 🦀 v1.39.0 with odzt60a
❯ ls
src  Cargo.toml

tmp/covtest is 📦 v0.1.0 via 🦀 v1.39.0 with odzt60a
❯ cargo tarpaulin
[INFO tarpaulin] Running Tarpaulin
[INFO tarpaulin] Building project
Error: "Failed to get test coverage! Error: Unable to parse binary."
[INFO tarpaulin] Launching test
[INFO tarpaulin] running /home/jlusby/tmp/covtest/target/debug/deps/covtest-d9ba10c7fe643f7e

tmp/covtest is 📦 v0.1.0 via 🦀 v1.39.0 with odzt60a
❯ cargo tarpaulin -v
[INFO tarpaulin] Running Tarpaulin
[INFO tarpaulin] Building project
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
[DEBUG tarpaulin] Processing Target(lib)
Error: "Failed to get test coverage! Error: Unable to parse binary."
[INFO tarpaulin] Launching test
[INFO tarpaulin] running /home/jlusby/tmp/covtest/target/debug/deps/covtest-d9ba10c7fe643f7e
```

**Comment by yaahc** [comment]
It's an x86_64 Intel i7 Dell laptop running Ubuntu 18.04

**Comment by yaahc** [comment]
Yup, that was it.

**Comment by xd009642** [comment]
So I just tried this with lld 8.0.0 and it worked fine. What lld version are you using? If it's 8 or newer if you upload the binary I can use objdump and other tools to try to figure out what's happening!

**Comment by yaahc** [comment]
I'm on lld 6.0.0

Here's the uploaded binary

[testcov-27f59cfc2b27f6da.gz](https://github.com/xd009642/tarpaulin/files/3887944/testcov-27f59cfc2b27f6da.gz)

**Comment by xd009642** [comment]
Hmm it all looks fine in objdump. Could you post the output of tarpaulin if you run it with the `--debug` flag?

**Comment by yaahc** [comment]
![Screenshot from 2019-11-26 13-27-49](https://user-images.githubusercontent.com/1993852/69674128-acc15180-1050-11ea-9467-123dae114490.png)

**Comment by xd009642** [comment]
Sorry it's been so long without any updates but I've revisited this issue today! 

I tried it with ubuntu 18.04, lld 6.0.0 the most recent tarpaulin release and rust 1.44.1 and everything worked perfectly. There's been a lot of big changes in the tarpaulin internals and potentially relevant changes in rust/cargo... Is this still an issue for you with lld? Just want to make sure there's nothing else that may cause failures
